### DISCUSSION TITLE
How to setup custom database notification trigger outside the panel

### DISCUSSION THREAD

**Discussion by discoverlance-com** [filamentphp/filament#14177] [Help] [answered]
### Package

Notifications

### Package Version

v3.2.110

### How can we help you?

In the filament documentation on [Rendering the database notifications modal](https://filamentphp.com/docs/3.x/notifications/database-notifications#rendering-the-database-notifications-modal):

**I quote as below:**

> If you'd like to render the database notifications modal outside of the [Panel Builder](https://filamentphp.com/docs/3.x/panels), you'll need to add a new Livewire component to your Blade layout:
> 
> @livewire('database-notifications')
> 
> To open the modal, you must have a "trigger" button in your view. Create a new trigger button component in your app, for instance at /resources/views/filament/notifications/database-notifications-trigger.blade.php:
> 
> ```blade
> <button type="button">
>     Notifications ({{ $unreadNotificationsCount }} unread)
> </button>
> 
> ```
> `$unreadNotificationsCount` is a variable automatically passed to this view, which provides it with a real-time count of unread notifications the user has.
> 
> In the service provider, point to this new trigger view:
> 
> ```php
> use Filament\Notifications\Livewire\DatabaseNotifications;
>  
> DatabaseNotifications::trigger('filament.notifications.database-notifications-trigger');
> 
> ```

This does not work. I have tried it and the trigger is not being set to my custom view. I get this error in my view.

```
Undefined variable $unreadNotificationsCount
```

I have added the `@livewire('database-notifications')` to my layout and created the same view as show in the example above but it does not work. Also, I have added the `DatabaseNotifications::trigger('filament.notifications.database-notifications-trigger');` to my `AppServiceProvider.php`. By the way, what does it mean by 

> In the service provider, point to this new trigger view:

Is it any service provider? I am not sure as it says, *the service provider**, Which method should it be added to, the `register` or `boot` method? For now, I added it to my boot method but I also tried it in the register method  and it's not working. Where should it be added because I think the issue might come from the trigger not being registered...

**Comment by leandrocfe** [comment, marked as answer]
create this file `resources/views/notifications-trigger.blade.php`

```html
<!-- resources/views/notifications-trigger.blade.php -->
<div>
    <div>
        <button type="button">
            Notifications ({{ $unreadNotificationsCount }} unread)
        </button>
    </div>
</div>
```

Then, add in the appserviceprovider

```php
public function boot(): void
{
    DatabaseNotifications::trigger('notifications-trigger');
}
```

**Reply by discoverlance-com** [reply]
I think I understand why it was not working. I misunderstood how the notifications trigger works. 

I assumed that I could add the livewire component, ` @livewire('database-notifications')` anywhere in my layout and then I could include the component or blade, `resources/views/notifications-trigger.blade.php` in another section of my page.

I was basically adding the livewire component in my layout and then using laravel blade `@include('notifications-trigger.blade.php')` to also include the component in my header. The idea was to add the notifications component to my header but I realised that my page header is also having a position relative so when I add the livewire component to the header, the slide out becomes relative to my header instead of the whole page. I thought I could use the method to first load the livewire component in the layout and then include the component it references anywhere on my page. It was an otherwise obvious mistake as `@livewire('some-livewire-component')` just directly adds the component wherever it's imported. Thanks.

Do you have any suggestions in this case though? Because my header is relative and it seems the slideout is being opened relative to the header height which is like 90px.

**Reply by leandrocfe** [reply]
What about move `database-notifications` component to your app.blade.php?

**Reply by discoverlance-com** [reply]
Hmmm.... I am not sure though how that could work. It will basically be the same thing right?

I basically have something like this

```blade
<nav class="sticky flex h-24 w-full select-none items-center" x-data="{ showMenu: false }">
    <div
        class="relative mx-auto flex w-full flex-wrap items-start justify-between font-medium md:h-24 md:items-center md:justify-between">
        <x-logo class="w-1/4 py-4 pl-6 pr-4 md:py-0" />
        <div :class="{ 'flex': showMenu, 'hidden md:flex': !showMenu }"
            class="absolute left-1/2 z-50 h-auto w-full -translate-x-1/2 flex-col items-center justify-center rounded-full border-0 border-gray-400 px-2 text-center text-gray-600 md:flex md:h-10 md:w-auto md:flex-row md:items-center md:border dark:border-gray-700 dark:text-gray-400">
            <a wire:navigate href="{{ route('home') }}"
                class="{{ $currentRouteName === 'home' ? 'dark:text-white text-gray-900' : 'dark:hover:text-white hover:text-gray-900' }} group relative mx-2 inline-block h-full w-full px-4 py-4 text-center font-medium leading-tight duration-300 ease-out md:w-auto md:px-2 md:py-2 md:text-center lg:mx-3">
                <span>Home</span>
                <span
                    class="{{ $currentRouteName === 'home' ? 'left-0  w-full' : 'left-1/2 w-0 group-hover:left-0 group-hover:w-full' }} absolute bottom-0 h-0.5 translate-y-px bg-gradient-to-r from-gray-900 via-gray-600 to-gray-900 duration-300 ease-out md:from-gray-700 md:via-gray-400 md:to-gray-700 dark:h-px"></span>
        </div>
        <div class="fixed left-0 top-0 z-40 hidden h-full w-full items-center bg-gray-200 bg-opacity-50 p-3 text-sm md:relative md:flex md:w-auto md:bg-transparent md:p-0 dark:bg-gray-900"
            :class="{ 'flex': showMenu, 'hidden': !showMenu }">
            <div
                class="h-full w-full select-none flex-col items-center overflow-hidden bg-white bg-opacity-50 p-3 backdrop-blur-lg md:relative md:flex md:h-auto md:flex-row md:overflow-auto md:bg-transparent md:p-0 dark:bg-black">
                <div class="flex h-full w-full flex-col items-center justify-end px-2 py-2 md:w-full md:flex-row">
                    @guest
                       <button>Sign In</button>
                    @endguest

                    @auth
                        @livewire('database-notifications')
                    @endauth
                </div>
            </div>
        </div>
        <div @click="showMenu = !showMenu"
            class="absolute right-0 z-50 mr-4 flex h-10 w-10 translate-y-1.5 cursor-pointer flex-col items-end rounded-full p-2 hover:bg-gray-400 hover:bg-opacity-10 md:hidden dark:hover:bg-gray-200/10"
            :class="{ 'text-gray-400': showMenu, 'text-gray-100': !showMenu }">
            <svg class="h-6 w-6 text-gray-600" x-show="!showMenu" fill="none" stroke-linecap="round"
                stroke-linejoin="round" stroke-width="2" viewBox="0 0 24 24" stroke="currentColor" x-cloak>
                <path d="M4 6h16M4 12h16M4 18h16"></path>
            </svg>
            <svg class="h-6 w-6" x-show="showMenu" fill="none" stroke="currentColor" viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg" x-cloak>
                <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M6 18L18 6M6 6l12 12">
                </path>
            </svg>
        </div>
    </div>
</nav>
```

So even if I move my component into the main blade layout,  it's still going to be the same I think as I only added it in the header blade file and added that component to the main layout. I need it to also be in the navigation so I can't really move it out of that context.
